src/relay/ir/module.cc

```diff
@@ -112,10 +112,13 @@ void ModuleNode::Add(const GlobalVar& var,
 }
 
 void ModuleNode::RegisterConstructors(const GlobalTypeVar& var, const TypeData& type) {
+  // We hash the global type var name to use as a globally unique prefix for tags.
+  // The hash will be used as the most significant byte of the tag, with the index of
+  // the constructor in the less significant bytes
+  size_t hash = std::hash<std::string>()(var->var->name_hint);
+  int64_t prefix = static_cast<int64_t>(hash & 0xff) << 56;
```

Comment:
Maybe use `int32`? I think that's more than enough.

Reply:
Probably, if we're truncating.